packages/x-tree-view/src/internals/plugins/useTreeViewKeyboardNavigation/useTreeViewKeyboardNavigation.ts

```diff
@@ -121,7 +121,15 @@ export const useTreeViewKeyboardNavigation: TreeViewPlugin<
       // If the focused item has children, we expand it.
       // If the focused item has no children, we select it.
       case key === 'Enter': {
-        if (canToggleItemExpansion(itemId)) {
+        if (
+          instance.isItemBeingEdited &&
+          instance.isItemEditable &&
+          instance.setEditedItemId &&
```

Comment:
there must be a better way of doing this 😿

Reply:
We could maybe add a check like `instance.hasPlugin(useTreeViewEditLabel)` that magically makes all the methods not nullable.

But until then I'm totally fine checking one method and then using `!` for all the others.

Reply:
I created the [issue ](https://github.com/mui/mui-x/issues/13777) because we will need this kind of magic solution 🙈 I have checks like this in multiple places and I don't like them 😓 
But I'll use `!` until then 🙈